ui/partials/main/intermittent.html

```diff
@@ -20,16 +20,31 @@
 
   <div id="failureSummaryGroup" class="collapsed">
     <label id="modalSummarylabel" for="modalSummary">Summary:</label>
-    <input id="modalSummary" type="text" placeholder="Intermittent..." maxlength="255" ng-model="modalSummary" />
+    <input id="modalSummary" type="text" placeholder="Intermittent..." pattern=".{0,255}" 
+           ng-model="modalSummary" ng-model-options="{allowInvalid:true}" />
+    <span id="modalSummaryLength" ng-bind="modalSummary.length" />
     <a ng-class="{'filersummary-open-btn': isFilerSummaryVisible}" prevent-default-on-left-click>
-      <i ng-click="toggleFilerSummaryVisibility()" ng-hide="isFilerSummaryVisible" class="fa fa-chevron-right" uib-tooltip="Show all failure lines for this job"></i>
+      <i ng-click="toggleFilerSummaryVisibility()" ng-hide="isFilerSummaryVisible" 
+         class="fa fa-chevron-right" uib-tooltip="Show all failure lines for this job">
+      </i>
       <span ng-show="isFilerSummaryVisible">
         <i ng-click="toggleFilerSummaryVisibility()" class="fa fa-chevron-down" uib-tooltip="Hide all failure lines for this job"></i>
         <textarea id="modalFailureList">{{thisFailure}}</textarea>
       </span>
     </a>
   </div>
 
+  <div id="dxrbuttons">
+    <a ng-show="hasPossibleFileName()" 
+       ng-href="https://dxr.mozilla.org/{{dxrTree}}/search?q=file%3A{{possibleFilename}}&redirect=true">DXR</a>
+    <a ng-show="isReleaseOrOlder()" 
+       ng-href="https://dxr.mozilla.org/mozilla-beta/search?q=file%3A{{possibleFilename}}&redirect=true">DXR (m-b)</a>
+    <a ng-show="isBetaOrOlder()" 
+       ng-href="https://dxr.mozilla.org/mozilla-aurora/search?q=file%3A{{possibleFilename}}&redirect=true">DXR (m-a)</a>
+    <a ng-show="isAuroraOrOlder()" 
+       ng-href="https://dxr.mozilla.org/mozilla-central/search?q=file%3A{{possibleFilename}}&redirect=true">DXR (m-c)</a>
```

Comment:
This is really hardcoded. The constants need to be moved elsewhere. And ideally we should make this more dynamic.